### PR TITLE
Make error messages more consistent in the AoC daystar view

### PR DESCRIPTION
**Pull request by D0rs4n**
## Description

Changes in the AoC DayAndStarView:
 - From now on, when the interacting user and the original author of the view is different, the bot will send an ephemeral message, regarding the issue, instead of the default error message.
## Did you:

- [x] Join the [**Python Discord Community**](https://discord.gg/python)?
- [x] Read all the comments in this template?
- [x] Ensure there is an issue open, or link relevant discord discussions?
- [x] Read and agree to the [contributing guidelines](https://pythondiscord.com/pages/contributing/contributing-guidelines/)?

### CODE REVIEW

**Review by ks129 — CHANGES_REQUESTED** [review]
It is not so high priority, so I think this should be solved with exception raising and handling in global error handler. This would remove possible future duplication.

**Comment by D0rs4n** [comment]
> It is not so high priority, so I think this should be solved with exception raising and handling in global error handler. This would remove possible future duplication.

I'm not sure how that would work in the interaction environment.
If an error is raised in the `interaction_check`, it'll call the `View.on_error`([See](https://discordpy.readthedocs.io/en/master/api.html?highlight=interaction_check#discord.ui.View.on_error)): `The default implementation prints the traceback to stderr.`
So that would have to be overriden as well, accordingly.
Hence, I believe it's easier, to just send an interaction response.
(You would have to do the same procedure on every View.)

File: bot/exts/events/advent_of_code/views/dayandstarview.py [hunk 1]
```diff
@@ -42,7 +42,10 @@ def generate_output(self) -> str:
 
     async def interaction_check(self, interaction: discord.Interaction) -> bool:
         """Global check to ensure that the interacting user is the user who invoked the command originally."""
-        return interaction.user == self.original_author
+        if interaction.user != self.original_author:
+            await interaction.response.send_message(":x: This is not your view to interact with!", ephemeral=True)
```
Comment (janine9vn) [inline bot/exts/events/advent_of_code/views/dayandstarview.py:46]: My only comment is that while this message makes sense for someone familiar with interactions and discord.py, it may not be clear to those who don't know what a "view" means.

We could maybe reword this to something like: "You can't interact with someone else's response. Please run the command yourself."

**Review by janine9vn — CHANGES_REQUESTED** [review]
Small change on the wording of the message, but otherwise thank you for the PR~

File: bot/exts/events/advent_of_code/views/dayandstarview.py (hunk 1 above)
Comment (D0rs4n) [reply bot/exts/events/advent_of_code/views/dayandstarview.py]: That's a valid point.
I've pushed a commit that makes the error message a bit more understandable for everyone!

**Review by janine9vn — APPROVED** [review]
beautiful, thank you!

**Review by wookie184 — APPROVED** [review]
Works nicely, very cool!
